tests/migration_test.go

```diff
@@ -1459,6 +1503,108 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
+				By("Create a VMI that will run root(default)")
+				vmi := createVMI()
+
+				By("Starting the VirtualMachineInstance")
+				// Resizing takes too long and therefor a warning is thrown
+				vmi = runVMIAndExpectLaunchIgnoreWarnings(vmi, 240)
+
+				By("Checking that the VirtualMachineInstance console has expected output")
+				Expect(loginFunc(vmi)).To(Succeed())
+
+				By("Checking that the launcher is running as root")
+				Expect(tests.GetIdOfLauncher(vmi)).To(Equal("0"))
+
+				tests.EnableFeatureGate(virtconfig.NonRoot)
+
+				By("Starting new migration and waiting for it to succeed")
+				migration := tests.NewRandomMigration(vmi.Name, vmi.Namespace)
+				migrationUID := tests.RunMigrationAndExpectCompletion(virtClient, migration, 340)
+
+				By("Verifying Second Migration Succeeeds")
+				tests.ConfirmVMIPostMigration(virtClient, vmi, migrationUID)
+
+				By("Checking that the launcher is running as qemu")
+				Expect(tests.GetIdOfLauncher(vmi)).To(Equal("107"))
+				By("Checking that the VirtualMachineInstance console has expected output")
+				Expect(loginFunc(vmi)).To(Succeed())
+
+				By("Deleting the VMI")
+				Expect(virtClient.VirtualMachineInstance(vmi.Namespace).Delete(vmi.Name, &metav1.DeleteOptions{})).To(Succeed())
+
+				By("Waiting for VMI to disappear")
+				tests.WaitForVirtualMachineToDisappearWithTimeout(vmi, 240)
+
+			},
+				table.Entry("with simple VMI", func() *v1.VirtualMachineInstance {
+					return tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+				}, console.LoginToAlpine),
+
+				table.Entry("with DataVolume", func() *v1.VirtualMachineInstance {
+					nfsPod, pvName := createPVC()
+
+					// Create fake DV and new PV&PVC of that name. Otherwise VM can't be created
+					dv := tests.NewRandomDataVolumeWithPVCSource(util.NamespaceTestDefault, pvName, util.NamespaceTestDefault, k8sv1.ReadWriteMany)
+					dv.Spec.PVC.Resources.Requests["storage"] = resource.MustParse(size)
+					_, err := virtClient.CdiClient().CdiV1beta1().DataVolumes(dv.Namespace).Create(context.Background(), dv, metav1.CreateOptions{})
+					Expect(err).To(BeNil())
+					tests.CreateNFSPvAndPvc(dv.Name, util.NamespaceTestDefault, size, libnet.GetPodIpByFamily(nfsPod, ipProtocol), os)
+
+					return tests.NewRandomVMIWithDataVolume(dv.Name)
+				}, console.LoginToAlpine),
+
+				table.Entry("with CD + CloudInit + SA + ConfigMap + Secret + DownwardAPI + Kernel Boot", func() *v1.VirtualMachineInstance {
+					return cDandCloudInitServiceAccountConfigMapSecretDownwardAPIKernelBoot()
+				}, console.LoginToFedora),
+
+				table.Entry("with PVC", func() *v1.VirtualMachineInstance {
+					nfsPod, pvName := createPVC()
+
+					tests.CreateNFSPvAndPvc(pvName, util.NamespaceTestDefault, size, libnet.GetPodIpByFamily(nfsPod, ipProtocol), os)
+
+					return tests.NewRandomVMIWithPVC(pvName)
+				}, console.LoginToAlpine),
+			)
+		})
```

Comment:
What would you think about a test with a virtiofs-enabled VMI, that makes sure the migration doesn't happen and the VMI stays alive? (which I assume is the expected behavior)

Reply:
Ah yeah, nevermind then